fairlearn/postprocessing/_threshold_optimizer.py

```diff
@@ -427,6 +260,167 @@ def _threshold_optimization_equalized_odds(sensitive_features, labels, scores, g
     return predicted_EO_by_sensitive_feature
 
 
+_SUPPORTED_CONSTRAINTS = {
+    DEMOGRAPHIC_PARITY: _threshold_optimization_demographic_parity,
+    EQUALIZED_ODDS: _threshold_optimization_equalized_odds}
+
+
+def _get_soft_predictions(estimator, X):
+    """Returns soft predictions of a classifier using either `predict_proba`
+    or `decision_function` methods.
+    """
+    if hasattr(estimator, "predict_proba"):
+        return estimator.predict_proba(X)
+    elif hasattr(estimator, "decision_function"):
+        return estimator.decision_function(X)
+    else:
+        raise ValueError("{} provides neither a `predict_proba` nor a "
+                            "`decision_function` method.".format(
+                                estimator.__class__.__name__
+                            ))
+
+
+class ThresholdOptimizer(ClassifierMixin, BaseEstimator):
```

Comment:
@MiroDudik may have thoughts on this. It should probably be a new kind of "RandomizedClassifierMixin"?

Reply:
I think we can hold off on this for the time being. But since you mention this: the idea would be to have a uniform approach to how we return randomized classifiers that are represented as mixtures over deterministic classifiers. This currently pops up only here and in `reductions.ExponentiatedGradient`

Reply:
By "hold off on this" do you mean "leave as is in this PR" or "don't change what was there before the PR"? I think you mean the former, but just trying to confirm

Reply:
Sorry for confusion. Leave as is in this PR!